.github/workflows/prbot.yml

```diff
@@ -1,18 +1,157 @@
 name: PR Robot
-on:
-- pull_request
-
+on: [pull_request]
 jobs:
-  pr-robot:
+  build:
```

Comment:
The `build` job is ran before all other jobs. After building everything, it creates a tarball of the build and uploads it as an artifact, which other jobs can download & extract.